packages/block-editor/src/components/block-types-list/index.native.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import {
+	FlatList,
+	View,
+	TouchableWithoutFeedback,
+	Dimensions,
+} from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { BottomSheet, InserterButton } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+const MIN_COL_NUM = 3;
+
+export default function BlockTypesList( { name, items, onSelect, listProps } ) {
```

Comment:
The logic for this component has been extracted from [`InserterSearchResults ` component](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter/search-results.native.js). This component is now used for rendering the block types in the different tabs of the inserter menu as well as in the search results.